runtime/ztests/expr/case-match.yaml

```
@@ -0,0 +1,18 @@
+spq: 'values case x when 1 then "foo" when 2 then "bar" else {y:12} end'
```

Nit: No need to quote this.

```suggestion
spq: values case x when 1 then "foo" when 2 then "bar" else {y:12} end
```